### PR TITLE
in-tree: Add check for empty secret/context on create/delete 

### LINKED ISSUE #953: in-tree: Error message for no secret/context should be more descriptive
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE

**What happened**:
Users were getting a confusing error message when they didn't pass the secret name/context name. 

**What you expected to happen**:
The error should be clear that an secretName/contextName was not found.

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <grant@portworx.com>

**What this PR does / why we need it**:
* Check for empty secret/context on create/delete 
* Return respective error messages.

**Which issue(s) this PR fixes** (optional)
Closes #953

**Special notes for your reviewer**:

### CODE REVIEW

File: api/server/middleware_auth.go [hunk 1]
```diff
@@ -73,6 +73,15 @@ func (a *authMiddleware) createWithAuth(w http.ResponseWriter, r *http.Request,
 		json.NewEncoder(w).Encode(&dcRes)
 		return
 	}
+	if secretName == "" || secretContext == "" {
```
Comment (adityadani) [inline api/server/middleware_auth.go]: It is ok for secretContext to be empty. So you can remove it from the check

File: api/server/middleware_auth.go [hunk 2]
```diff
@@ -206,6 +215,14 @@ func (a *authMiddleware) deleteWithAuth(w http.ResponseWriter, r *http.Request,
 		json.NewEncoder(w).Encode(volumeResponse)
 		return
 	}
+	if secretName == "" || secretContext == "" {
+		errorMessage := "Internal error, no secret found in the annotations of the persistent volume claim"
```
Comment (adityadani) [inline api/server/middleware_auth.go]: @lpabon If a cluster goes from no auth -> auth, then all the Kubernetes requests are going to get this error. And we will not be able to delete a volume through Kubernetes. Is it ok to just doc this edge case?

File: api/server/middleware_auth.go [hunk 3]
```diff
@@ -206,6 +215,14 @@ func (a *authMiddleware) deleteWithAuth(w http.ResponseWriter, r *http.Request,
 		json.NewEncoder(w).Encode(volumeResponse)
 		return
 	}
+	if secretName == "" || secretContext == "" {
```
Comment (adityadani) [inline api/server/middleware_auth.go]: again no need to check for secretContext

File: api/server/middleware_auth.go (hunk 2 above)
Comment (lpabon) [reply api/server/middleware_auth.go]: offline discussion we decided that it is ok to fail the PVC delete request coming from Kubernetes, but provide the SDK as the method of cleaning up the volumes in this situation.

**Pull request merged**